optuna/storages/_journal/redis.py

```diff
@@ -91,5 +100,17 @@ def append_logs(self, logs: List[Dict[str, Any]]) -> None:
                 log_number = self._redis.incr(f"{self._prefix}:log_number", 1)
                 self._redis.set(self._key_log_id(log_number), json.dumps(log))
 
+    def save_snapshot(self, snapshot: bytes) -> None:
+        self._redis.set(f"{self._prefix}:snapshot", snapshot)
+
+    def load_snapshot(self, loader: Callable[[bytes], None]) -> None:
```

Comment:
Let me suggest the following change:
```suggestion
    def load_snapshot(self) -> bytes:
```
IMO, this interface is more intuitive as it's consistent with the signature of `save_snapshot()`, and simpler than the current one (taking `Callback`).

Reply:
Sounds good to me.

@c-bata Do you have any opinions for this? The current code is based on this suggestion.

Reply:
Thank you for the suggestion. The loader callback based interface is flexible, but it obviously sacrifices the simplicity, which is a strength of JournalStorage. So the suggestion sounds reasonable to me.

@HideakiImamura Could you also remove a `SnapshotRestoreError` exception class?

Reply:
Thanks for the comment. I would like to be sure how I should handle exceptions. For example, if an error occurs during unpickle of a snapshot. I don't think we should raise an error and stop the execution, can I just ignore this error? Other direction is how about generating a warning?

Reply:
Sounds good 👍  You can ignore any exceptions while restoring the snapshot and show warning messages.

Reply:
👍  > You can ignore any exceptions while restoring the snapshot and show warning messages.

Reply:
Thanks. I removed the `RestoreSnapshotError` and fixed to show warning messages when the loading snapshot fails. PTAL.